src/addons/Portal/Portal.js

```diff
@@ -146,6 +147,33 @@ function Portal(props) {
   // Component Event Handlers
   // ----------------------------------------
 
+  React.useEffect(() => {
+    if (!hideOnScroll) {
+      return
+    }
+    const abortController = new AbortController()
+
+    window.addEventListener(
+      'scroll',
+      (e) => {
+        debug('handleHideOnScroll()')
+
+        // Do not hide the popup when scroll comes from inside the popup
+        // https://github.com/Semantic-Org/Semantic-UI-React/issues/4305
+        if (_.isElement(e.target) && contentRef.current.contains(e.target)) {
+          return
+        }
+
+        closePortal(e)
+      },
+      { signal: abortController.signal, passive: true },
+    )
+
+    return () => {
+      abortController.abort()
+    }
+  }, [closePortal, hideOnScroll])
```

Comment:
As now we need to care about `closePortal` being stable, let's wrap it with `useEventCallback()` i.e.

```diff
- const closePortal = (e) => {
+ const closePortal = useEventCallback((e) => { 
    debug('close()')

    setOpen(false)
    _.invoke(props, 'onClose', e, { ...props, open: false })
- }
+ }) 
```